### PR TITLE
Don't propagate JavaInfo for compiler plugin deshaded libs

### PR DESCRIPTION
**Pull request by nkoroste**
If JavaInfo is propagated, compiler plugin libs end up in the set of runtime dependencies of android_binary targets, which makes no sense.
Fixes https://github.com/bazelbuild/rules_kotlin/issues/639

### CODE REVIEW

**Comment by nkoroste** [comment]
cc @mauriciogg

File: kotlin/internal/jvm/impl.bzl [hunk 1]
```diff
@@ -419,7 +419,7 @@ def kt_compiler_plugin_impl(ctx):
 
     return [
         DefaultInfo(files = classpath),
-        info,
+        java_common.merge([]),
```
Comment (nkoroste) [inline kotlin/internal/jvm/impl.bzl:422]:
```suggestion
```

**Review by Bencodes — APPROVED** [review]

File: kotlin/internal/jvm/impl.bzl (hunk 1 above)
Comment (restingbull) [inline kotlin/internal/jvm/impl.bzl:422]: What consumes this JavaInfo?

Comment (nkoroste) [reply kotlin/internal/jvm/impl.bzl:422]: I couldn't find anything

Comment (restingbull) [reply kotlin/internal/jvm/impl.bzl:422]: Let's remove it, then.

Comment (nkoroste) [reply kotlin/internal/jvm/impl.bzl]: sorry, updated

Comment (nkoroste) [reply kotlin/internal/jvm/impl.bzl]: Looking closer into it, I see it's referenced in
https://sourcegraph.com/github.com/bazelbuild/rules_kotlin/-/blob/kotlin/internal/jvm/compile.bzl?L707#tab=history
https://sourcegraph.com/github.com/bazelbuild/rules_kotlin/-/blob/kotlin/internal/jvm/plugins.bzl?L51

Do we still want to remove it?

Comment (restingbull) [reply kotlin/internal/jvm/impl.bzl]: Yes -- providing an empty JavaInfo doesn't help, and increases the chances of something stupid happening.

Speaking of stupid... why does the target have a `KtJvmPluginInfo` and no `JavaInfo`? It isn't due to `KtCompilerPluginInfo`.

Maybe it's an export issue?

**Review by restingbull — APPROVED** [review]

**Pull request merged**